kitsune/search/v2/base.py

```diff
@@ -48,13 +52,28 @@ def prepare(cls, instance):
                     # set is_dst=False to avoid errors when an ambiguous time is sent:
                     # https://docs.djangoproject.com/en/2.2/ref/utils/#django.utils.timezone.make_aware
                     value = timezone.make_aware(value, is_dst=False).astimezone(timezone.utc)
+
                 setattr(obj, f, value)
 
+        obj.updated_on = datetime.now(timezone.utc)
         obj.meta.id = instance.pk
-        obj.indexed_on = datetime.now(timezone.utc)
 
         return obj
 
+    def to_action(self, action=None, bulk=False, **kwargs):
+        """Method to construct the data for save, delete, update operations.
+
+        Useful for bulk operations.
+        """
+
+        # Default to index if no action is defined or if it's `save`
+        if bulk:
+            return self.to_dict(include_meta=True)
+        if not action or action == "index":
+            self.save(**kwargs)
+        elif action == "delete":
+            self.delete()
```

Comment:
My gut feels a little uncomfortable about this doing two very different things on the basis of the `bulk` argument: either returning a dictionary, or performing operations in ES. Since we only call this in the `index_object(s_bulk)` tasks - and I can't see us wanting or needing to index stuff outside of calling those tasks - do we need this abstraction?